src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive =[]
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
 
+    if (this.health) {
+      Animal.alive.push(this);
+    }
+  }
+}
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
-
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+  bite(food) {
+    if (!(food instanceof Carnivore || food.hidden)) {
+      food.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(x => (x.health > 0));
```

Comment:
Hm, I suggest to add this condition before: `if (food.health <= 0)`. So your code will be like this:
```suggestion
    if (food.health <= 0) {
      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
    }
```

Why? After bite you will iterate by your alive animals always. Even if your animal not die

Reply:
oh, I understand